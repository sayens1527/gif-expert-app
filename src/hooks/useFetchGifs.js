import { useEffect, useState } from "react"
import getGifs from "../helpers/Gifts";


export const useFetchGifs = (category) => {

    const [state, setstate] = useState({data:[], isLoading: true});

    useEffect(() => {
        getGifs(category).then(images => {
            setstate({data:images, isLoading:false});
        });
    }, [category]);


    return state;
    
}
