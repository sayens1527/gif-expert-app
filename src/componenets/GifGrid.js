import React from 'react'
import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ( { category } ) => {

    const {data:images, isLoading} =  useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>
            {isLoading && <p>Loading</p> }
            <div className='grid'>
                {   
                    images.map( img => (
                        <GifGridItem  
                            key={img.id}
                            {...img}/>
                        
                    ))   
                }
            </div>
        </>
    )
}

export default GifGrid
