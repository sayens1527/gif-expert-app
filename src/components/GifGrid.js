import React from 'react'
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ( { category } ) => {

    const {data:images, isLoading} =  useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>
            {isLoading && <p>Loading</p> }
            <div className='grid'>
                {   
                    images.map( img => (
                        <GifGridItem  
                            key={img.id}
                            {...img}/>
                        
                    ))   
                }
            </div>
        </>
    )
}

GifGrid.propTypes =  {
    category:PropTypes.string.isRequired
}

export default GifGrid
