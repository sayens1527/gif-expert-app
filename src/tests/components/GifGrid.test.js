import {shallow} from 'enzyme';
import GifGrid from "../../components/GifGrid";
import '@testing-library/jest-dom';
import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('test in GifGrid component', () => {

    const category = 'One punch';


    test('should match with snapshot', () => {

        useFetchGifs.mockReturnValue({data:[],isLoading:true});

        const wraper = shallow(<GifGrid category= { category }></GifGrid>);
        
        expect(wraper).toMatchSnapshot();
    });
    
    test('should show the images when use useFecthGifts', () => {

        const gifs =  [
            {
                id: '123',
                title:'titulo',
                url:'https://www.algo.com/algo.png'
            },
            {
                id: '1234',
                title:'titulo',
                url:'https://www.algo.com/algo.png'
            },
            {
                id: '12345',
                title:'titulo',
                url:'https://www.algo.com/algo.png'
            }
        ]

        useFetchGifs.mockReturnValue({data:gifs,isLoading:false});

        const wraper = shallow(<GifGrid category= { category }></GifGrid>);

        const loading = wraper.find('p');
        const gifGridItems = wraper.find('GifGridItem').length;

        expect(gifGridItems).toBe(gifs.length);
        expect(loading.exists()).toBe(false);

    });
    
});
