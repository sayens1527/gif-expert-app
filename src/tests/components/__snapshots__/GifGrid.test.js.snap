// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifGrid component should match with snapshot 1`] = `
<Fragment>
  <h3>
     
    One punch
     
  </h3>
  <p>
    Loading
  </p>
  <div
    className="grid"
  />
</Fragment>
`;
